TESTS WITH ERRORS 
test03_GM_Hanoi_03 (gradingtests.UnitTests)
Traceback (most recent call last):
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 108, in test03_GM_Hanoi_03
    th.makeMove(movables[0])
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\student_code_game_masters.py", line 96, in makeMove
    if len(disks_on_target_peg)==0:
TypeError: object of type 'int' has no len()

test04_GM_Hanoi_04 (gradingtests.UnitTests)
Traceback (most recent call last):
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 123, in test04_GM_Hanoi_04
    th.makeMove(movables[0])
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\student_code_game_masters.py", line 96, in makeMove
    if len(disks_on_target_peg)==0:
TypeError: object of type 'int' has no len()

test09_DFS_Hanoi_01 (gradingtests.UnitTests)
Traceback (most recent call last):
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 227, in test09_DFS_Hanoi_01
    [14, ((), (), (1, 2, 3))],
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 52, in runPlayXSteps
    results = self.pool.apply_async(self.playXSteps, [solver, plays]).get(timeout)
  File "C:\Program Files\Python37\lib\multiprocessing\pool.py", line 683, in get
    raise self._value
  File "C:\Program Files\Python37\lib\multiprocessing\pool.py", line 121, in worker
    result = (True, func(*args, **kwds))
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 27, in playXSteps
    solver.solveOneStep()
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\student_code_uninformed_solvers.py", line 45, in solveOneStep
    self.gm.makeMove(next_move)
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\student_code_game_masters.py", line 81, in makeMove
    disks_on_target_peg = self.checkOnPeg(target_peg)[0]
IndexError: list index out of range

test10_DFS_Hanoi_02 (gradingtests.UnitTests)
Traceback (most recent call last):
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 246, in test10_DFS_Hanoi_02
    [20, ((2,5,), (4,), (1, 3))],
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 52, in runPlayXSteps
    results = self.pool.apply_async(self.playXSteps, [solver, plays]).get(timeout)
  File "C:\Program Files\Python37\lib\multiprocessing\pool.py", line 683, in get
    raise self._value
  File "C:\Program Files\Python37\lib\multiprocessing\pool.py", line 121, in worker
    result = (True, func(*args, **kwds))
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 27, in playXSteps
    solver.solveOneStep()
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\student_code_uninformed_solvers.py", line 45, in solveOneStep
    self.gm.makeMove(next_move)
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\student_code_game_masters.py", line 81, in makeMove
    disks_on_target_peg = self.checkOnPeg(target_peg)[0]
IndexError: list index out of range

test11_DFS_Hanoi_03 (gradingtests.UnitTests)
Traceback (most recent call last):
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 260, in test11_DFS_Hanoi_03
    self.runSolve(solver)
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 68, in runSolve
    self.pool.apply_async(self.solve, [solver,]).get(timeout)
  File "C:\Program Files\Python37\lib\multiprocessing\pool.py", line 683, in get
    raise self._value
  File "C:\Program Files\Python37\lib\multiprocessing\pool.py", line 121, in worker
    result = (True, func(*args, **kwds))
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 40, in solve
    solver.solve()
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\solver.py", line 45, in solve
    while not self.solveOneStep():
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\student_code_uninformed_solvers.py", line 45, in solveOneStep
    self.gm.makeMove(next_move)
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\student_code_game_masters.py", line 81, in makeMove
    disks_on_target_peg = self.checkOnPeg(target_peg)[0]
IndexError: list index out of range

test12_BFS_Hanoi_01 (gradingtests.UnitTests)
Traceback (most recent call last):
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 278, in test12_BFS_Hanoi_01
    [22, ((2,), (3,), (1,))],
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 52, in runPlayXSteps
    results = self.pool.apply_async(self.playXSteps, [solver, plays]).get(timeout)
  File "C:\Program Files\Python37\lib\multiprocessing\pool.py", line 683, in get
    raise self._value
  File "C:\Program Files\Python37\lib\multiprocessing\pool.py", line 121, in worker
    result = (True, func(*args, **kwds))
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 27, in playXSteps
    solver.solveOneStep()
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\student_code_uninformed_solvers.py", line 96, in solveOneStep
    if self.solveOneStep():
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\student_code_uninformed_solvers.py", line 115, in solveOneStep
    self.gm.makeMove(m)
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\student_code_game_masters.py", line 81, in makeMove
    disks_on_target_peg = self.checkOnPeg(target_peg)[0]
IndexError: list index out of range

test13_BFS_Hanoi_02 (gradingtests.UnitTests)
Traceback (most recent call last):
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 296, in test13_BFS_Hanoi_02
    [23, ((5,), (4,), (1, 2, 3))],
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 52, in runPlayXSteps
    results = self.pool.apply_async(self.playXSteps, [solver, plays]).get(timeout)
  File "C:\Program Files\Python37\lib\multiprocessing\pool.py", line 683, in get
    raise self._value
  File "C:\Program Files\Python37\lib\multiprocessing\pool.py", line 121, in worker
    result = (True, func(*args, **kwds))
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 27, in playXSteps
    solver.solveOneStep()
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\student_code_uninformed_solvers.py", line 96, in solveOneStep
    if self.solveOneStep():
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\student_code_uninformed_solvers.py", line 115, in solveOneStep
    self.gm.makeMove(m)
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\student_code_game_masters.py", line 96, in makeMove
    if len(disks_on_target_peg)==0:
TypeError: object of type 'int' has no len()

test14_BFS_Hanoi_03 (gradingtests.UnitTests)
Traceback (most recent call last):
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 314, in test14_BFS_Hanoi_03
    [23, ((1, 4, 5), (), (2, 3))],
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 52, in runPlayXSteps
    results = self.pool.apply_async(self.playXSteps, [solver, plays]).get(timeout)
  File "C:\Program Files\Python37\lib\multiprocessing\pool.py", line 683, in get
    raise self._value
  File "C:\Program Files\Python37\lib\multiprocessing\pool.py", line 121, in worker
    result = (True, func(*args, **kwds))
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 27, in playXSteps
    solver.solveOneStep()
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\student_code_uninformed_solvers.py", line 96, in solveOneStep
    if self.solveOneStep():
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\student_code_uninformed_solvers.py", line 115, in solveOneStep
    self.gm.makeMove(m)
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\student_code_game_masters.py", line 96, in makeMove
    if len(disks_on_target_peg)==0:
TypeError: object of type 'int' has no len()

test15_DFS_8Puzzle_01 (gradingtests.UnitTests)
Traceback (most recent call last):
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 333, in test15_DFS_8Puzzle_01
    [20, ((-1, 1, 4), (5, 2, 3), (6, 7, 8))],
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 52, in runPlayXSteps
    results = self.pool.apply_async(self.playXSteps, [solver, plays]).get(timeout)
  File "C:\Program Files\Python37\lib\multiprocessing\pool.py", line 683, in get
    raise self._value
  File "C:\Program Files\Python37\lib\multiprocessing\pool.py", line 121, in worker
    result = (True, func(*args, **kwds))
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 27, in playXSteps
    solver.solveOneStep()
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\student_code_uninformed_solvers.py", line 49, in solveOneStep
    while (((state.parent is not None) and (self.gm.getGameState() == state.parent.state))) or GameState(self.gm.getGameState(), 0, None) in self.visited:
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\student_code_game_masters.py", line 153, in getGameState
    num3 = bindings_lst[0].bindings[0].constant.element
TypeError: 'bool' object is not subscriptable

test16_DFS_8Puzzle_02 (gradingtests.UnitTests)
Traceback (most recent call last):
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 352, in test16_DFS_8Puzzle_02
    [30, ((8, 1, -1), (7, 2, 3), (6, 5, 4))],
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 52, in runPlayXSteps
    results = self.pool.apply_async(self.playXSteps, [solver, plays]).get(timeout)
  File "C:\Program Files\Python37\lib\multiprocessing\pool.py", line 683, in get
    raise self._value
  File "C:\Program Files\Python37\lib\multiprocessing\pool.py", line 121, in worker
    result = (True, func(*args, **kwds))
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 27, in playXSteps
    solver.solveOneStep()
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\student_code_uninformed_solvers.py", line 49, in solveOneStep
    while (((state.parent is not None) and (self.gm.getGameState() == state.parent.state))) or GameState(self.gm.getGameState(), 0, None) in self.visited:
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\student_code_game_masters.py", line 159, in getGameState
    num6 = bindings_lst[0].bindings[0].constant.element
TypeError: 'bool' object is not subscriptable

test17_DFS_8Puzzle_03 (gradingtests.UnitTests)
Traceback (most recent call last):
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 371, in test17_DFS_8Puzzle_03
    [8, ((8, 3, 4), (2, -1, 1), (7, 6, 5))],
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 52, in runPlayXSteps
    results = self.pool.apply_async(self.playXSteps, [solver, plays]).get(timeout)
  File "C:\Program Files\Python37\lib\multiprocessing\pool.py", line 683, in get
    raise self._value
  File "C:\Program Files\Python37\lib\multiprocessing\pool.py", line 121, in worker
    result = (True, func(*args, **kwds))
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 27, in playXSteps
    solver.solveOneStep()
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\student_code_uninformed_solvers.py", line 49, in solveOneStep
    while (((state.parent is not None) and (self.gm.getGameState() == state.parent.state))) or GameState(self.gm.getGameState(), 0, None) in self.visited:
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\student_code_game_masters.py", line 159, in getGameState
    num6 = bindings_lst[0].bindings[0].constant.element
TypeError: 'bool' object is not subscriptable

test18_BFS_8Puzzle_01 (gradingtests.UnitTests)
Traceback (most recent call last):
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 390, in test18_BFS_8Puzzle_01
    [21, ((6, 5, 4), (1, -1, 8), (7, 3, 2))],
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 52, in runPlayXSteps
    results = self.pool.apply_async(self.playXSteps, [solver, plays]).get(timeout)
  File "C:\Program Files\Python37\lib\multiprocessing\pool.py", line 683, in get
    raise self._value
  File "C:\Program Files\Python37\lib\multiprocessing\pool.py", line 121, in worker
    result = (True, func(*args, **kwds))
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 27, in playXSteps
    solver.solveOneStep()
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\student_code_uninformed_solvers.py", line 116, in solveOneStep
    if (((state.parent is not None) and (self.gm.getGameState() == state.parent.state))) or GameState(self.gm.getGameState(), 0, None) in self.visited:
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\student_code_game_masters.py", line 151, in getGameState
    num2 = bindings_lst[0].bindings[0].constant.element
TypeError: 'bool' object is not subscriptable

test19_BFS_8Puzzle_02 (gradingtests.UnitTests)
Traceback (most recent call last):
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 409, in test19_BFS_8Puzzle_02
    [6, ((5, 4, 8), (6, 1, 2), (7, 3, -1))],
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 52, in runPlayXSteps
    results = self.pool.apply_async(self.playXSteps, [solver, plays]).get(timeout)
  File "C:\Program Files\Python37\lib\multiprocessing\pool.py", line 683, in get
    raise self._value
  File "C:\Program Files\Python37\lib\multiprocessing\pool.py", line 121, in worker
    result = (True, func(*args, **kwds))
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 27, in playXSteps
    solver.solveOneStep()
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\student_code_uninformed_solvers.py", line 116, in solveOneStep
    if (((state.parent is not None) and (self.gm.getGameState() == state.parent.state))) or GameState(self.gm.getGameState(), 0, None) in self.visited:
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\student_code_game_masters.py", line 151, in getGameState
    num2 = bindings_lst[0].bindings[0].constant.element
TypeError: 'bool' object is not subscriptable

test20_BFS_8Puzzle_03 (gradingtests.UnitTests)
Traceback (most recent call last):
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 428, in test20_BFS_8Puzzle_03
    [7, ((1, 2, -1), (8, 4, 3), (7, 6, 5))],
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 52, in runPlayXSteps
    results = self.pool.apply_async(self.playXSteps, [solver, plays]).get(timeout)
  File "C:\Program Files\Python37\lib\multiprocessing\pool.py", line 683, in get
    raise self._value
  File "C:\Program Files\Python37\lib\multiprocessing\pool.py", line 121, in worker
    result = (True, func(*args, **kwds))
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 27, in playXSteps
    solver.solveOneStep()
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\student_code_uninformed_solvers.py", line 116, in solveOneStep
    if (((state.parent is not None) and (self.gm.getGameState() == state.parent.state))) or GameState(self.gm.getGameState(), 0, None) in self.visited:
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\student_code_game_masters.py", line 159, in getGameState
    num6 = bindings_lst[0].bindings[0].constant.element
TypeError: 'bool' object is not subscriptable

-------------------------------
TESTS FAILED  
test05_GM_8Puzzle_01 (gradingtests.UnitTests)
Traceback (most recent call last):
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 136, in test05_GM_8Puzzle_01
    self.assertTrue(p8.isWon())
AssertionError: False is not true

test07_GM_8Puzzle_03 (gradingtests.UnitTests)
Traceback (most recent call last):
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 175, in test07_GM_8Puzzle_03
    self.assertEqual(p8.getGameState(), ((5,1,4), (6,-1,8), (7,3,2)))
AssertionError: Tuples differ: ((5, 4, -1), (6, 1, 8), (7, 3, 2)) != ((5, 1, 4), (6, -1, 8), (7, 3, 2))

First differing element 0:
(5, 4, -1)
(5, 1, 4)

- ((5, 4, -1), (6, 1, 8), (7, 3, 2))
?       ----

+ ((5, 1, 4), (6, -1, 8), (7, 3, 2))
?      +++        +


test08_GM_8Puzzle_04 (gradingtests.UnitTests)
Traceback (most recent call last):
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 200, in test08_GM_8Puzzle_04
    self.assertEqual(p8.getGameState(), ((1,-1,3,),(8,2,4),(7,6,5)))
AssertionError: Tuples differ: ((1, 2, 3), (8, -1, 4), (7, 6, 5)) != ((1, -1, 3), (8, 2, 4), (7, 6, 5))

First differing element 0:
(1, 2, 3)
(1, -1, 3)

- ((1, 2, 3), (8, -1, 4), (7, 6, 5))
?      ^          ^^

+ ((1, -1, 3), (8, 2, 4), (7, 6, 5))
?      ^^          ^


-------------------------------
TESTS SKIPPED 
-------------------------------
